### DISCUSSION TITLE
To Run and Build

### DISCUSSION THREAD

**Discussion by Akshayp02** [jayasuryat/mendable#32] [Q&A] [answered]
![image](https://github.com/jayasuryat/mendable/assets/95123298/1d7f618b-eff6-42bb-9a9e-1abb6e30577d)

@jayasuryat  I have changed the code by for #31 but i'm not able to see it in run time please can you guide me about this.

**Comment by jayasuryat** [comment, marked as answer]
Hey @Akshayp02,

Could you please elaborate on what you mean by 
> i'm not able to see it in run time

Do you mean you are not able to run it?

---

On other note, the expectation here is to change the `filename : String` to `file : File`
I've added details about this in the TLDR part of the [original comment](https://github.com/jayasuryat/mendable/issues/31#issue-1861910787).

> ### TLDR:
> So, instead of using `filename` to parse a `Module` object, it would be better to parse it based on the `File` itself.
> 
> A `File` already has the `filename`, so all of the existing logics are covered, and it also has `path` which could be used further nail down module name more appropriately, especially in cases like the one described above (by comparing the file path with the root project's path for example).

---

Feel free to ask for any clarification if anything that I said above isn't clear.

**Reply by Akshayp02** [reply]
Yes i understand the task but I'm not able to run the main program to see the changes I made. Please can you guide me.

**Reply by jayasuryat** [reply]
Sure thing @Akshayp02,

###### I actually had set-up a file based [run configuration](https://github.com/jayasuryat/mendable/blob/a7ee7c9b129b5849fcfa55f259794281a5b7ea49/.run/MainKt.run.xml#L1-L10) so that anyone who has cloned the project would be able to run it. But looks like (from the screenshot which you have attached) it is not working as intended. I'm not sure if this is an IDE version issue or if it is something else which I'm missing.

---

Anyways,

You can run the project from [`com.jayasuryat.mendable.app.main`](https://github.com/jayasuryat/mendable/blob/a7ee7c9b129b5849fcfa55f259794281a5b7ea49/mendable-app/src/main/kotlin/com/jayasuryat/mendable/app/Main.kt#L36). It is in the `:mendable-app` module, filename `Main.kt`.

There should be a `run` button just beside the `main` function. Which you should be able to click then select `Run Main.kt` to run the project.
<img width="801" alt="1" src="https://github.com/jayasuryat/mendable/assets/37530409/c66e816f-ccd3-447f-bc1e-e085fd820689">

Make sure you point the program to a directory which contains the generated `Compose Compiler Metrics`. Here is how you can configure that:
<img width="801" alt="2" src="https://github.com/jayasuryat/mendable/assets/37530409/4de9eeaf-285d-4f83-a7c6-693eaecf3c01">

You can click on `Modify run configuration` and then give it the path which has the generated `Compose Compiler Metrics`.

<img width="667" alt="3" src="https://github.com/jayasuryat/mendable/assets/37530409/3e2375fb-e97e-4c4e-8619-fd26f0383781">

That should do it.

---

Adding this just in case, this is [how you generate Compose Compiler Metrics](https://jayasuryat.github.io/mendable/others/HowToGenerate/). (This assumes you have some project with Jetpack Compose, with which you can generate these metrics).

Let me know if you need any further information about anything.

**Reply by Akshayp02** [reply]
Okay sure, I'll follow these steps. Thank you so much for taking time and sending for each step. 😀
